mmdet/datasets/pipelines/transforms.py

```diff
@@ -1964,16 +1964,20 @@ class Mosaic:
            image. Default to (640, 640).
         center_ratio_range (Sequence[float]): Center ratio range of mosaic
            output. Default to (0.5, 1.5).
+        min_bbox_size (int | float): The minimum pixel for filtering
+            invalid bboxes after the mosaic pipeline. Default to 1.
         pad_val (int): Pad value. Default to 114.
     """
 
     def __init__(self,
                  img_scale=(640, 640),
                  center_ratio_range=(0.5, 1.5),
+                 min_bbox_size=1,
```

Comment:
In order not to affect the original logic, the default value should be set to 0. And only execute when it is greater than 0.

Reply:
Thank you for your kindly reminding and advice.

The added filtering rule aims to fix the BUG of NAN loss during traning. 
## The original code
In Mosaic, the sub image are cropped if the sub image is larger than mosaic patch. However, some bboxes are translated out of image completely (such as bbox2 in figure 1) and some bboxes are translated out of image partly  (such as bbox1 in figure 1).
 ```python
             figure 1. before clip 
            +----------------------+
            |                      |
            |                      |    
            |                      |   +----+
            |                      |   |    |
            |    +------+          |   +----+
            |    |      |          |    bbox2
            +----+------+----------+
                 |      |
                 +------+ 
                  bbox1
```
The internal method _mosaic_transform use following code to deal with it.
```python
            mosaic_bboxes[:, 0::2] = np.clip(mosaic_bboxes[:, 0::2], 0,
                                             2 * self.img_scale[1])
            mosaic_bboxes[:, 1::2] = np.clip(mosaic_bboxes[:, 1::2], 0,
                                             2 * self.img_scale[0])
```
After that, the bbox1 are cropped correctly, but the bbox2 are clipped into a line as shown in figure 2.
```python
             figure 2. before clip 
            +----------------------+
            |                      |
            |                      |    
            |                      + 
            |                      | bbox2  
            |    +------+          + 
            |    |      |          |
            +----+------+----------+
                  bbox1
```
That will lead to NaN loss during training.
## The modified code
The adding filtering rule aims to filter out the lines (such bbox2 in figure 2) to avoid NaN loss. So I have changed the default value to 0 but it will be executed when it is equal to 0.

Reply:
@hhaAndroid  The above reply is the reason why the new code execute when the threshold is equal to 0.

Reply:
Thank you for your detailed answer. Are you interested in adding the above content as docstr to the filtering rules?

Reply:
Thank you for your advices. The above content does not need to be docstr because it describes how the bug occurs and the bug have been fixed.

Reply:
If you are free to split the above content into 2 PRs, the first bug will mention a PR separately, and then we will merge it today.

Reply:
Thank you very much

Reply:
Thank you very much, I have modified the file and remove the fix of first bug in this PR at commit "fix this bug in another PR".